### PR TITLE
[bug] Fix patterns filtering 

### PR DESCRIPTION
**Pull request by sudhanshutech**
**Notes for Reviewers**

This PR fixes #

https://github.com/meshery/meshery/assets/90546692/f454029b-6e1b-404b-8ff7-7af0d8504b22

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by sudhanshutech** [comment]
Do not merge because I have to test a particular criteria then i will remove

**Comment by leecalcote** [comment]
@sudhanshutech when you perform a search for a term that only matches a published entry, does it show up?

**Comment by sudhanshutech** [comment]
> @sudhanshutech when you perform a search for a term that only matches a published entry, does it show up?

image

**Review by leecalcote — APPROVED** [review]
Thanks for confirming.

File: ui/components/MesheryPatterns.js [hunk 1]
```diff
@@ -634,30 +634,30 @@ function MesheryPatterns({
...
-        initPatternsSubscription();
-      },
-      error: (err) => console.log('There was an error fetching Catalog Filter: ', err),
-    });
-
-    return () => {
-      fetchCatalogPatterns.unsubscribe();
-      disposeConfSubscriptionRef.current?.dispose();
-    };
+    // const fetchCatalogPatterns = fetchCatalogPattern({
```
Comment (leecalcote) [inline ui/components/MesheryPatterns.js:637]: @sudhanshutech will you document why this section remains commented and not deleted?

It's fine that it is not deleted, but document why is it commented out and why it's still around, please.

File: ui/components/MesheryPatterns.js [hunk 2]
```diff
@@ -634,30 +634,37 @@ function MesheryPatterns({
...
 
-    return () => {
-      fetchCatalogPatterns.unsubscribe();
-      disposeConfSubscriptionRef.current?.dispose();
-    };
+    /*
+     Below is a graphql query that fetches the catalog patterns that is published so
+     when catalogVisibility is true, we fetch the catalog patterns and set it to the patterns state
+     which show the catalog patterns only in the UI at the top of the list always whether we filter for public or private patterns.
+     And our rest api already fetch published so for we not rquired this.
```
Comment (leecalcote) [inline ui/components/MesheryPatterns.js:642]:
```suggestion
     Meshery's REST API already fetches catalog items with `published` visibility, hence this function is commented out.
```

Comment (leecalcote) [reply ui/components/MesheryPatterns.js:642]: Thanks, @sudhanshutech. Now: why are we leaving the commented code in the code base?

**Pull request merged**